onnxruntime/core/providers/nuphar/scripts/symbolic_shape_infer.py

```diff
@@ -777,9 +777,12 @@ def _infer_NonZero(self, node):
 
     def _infer_OneHot(self, node):
         shape = self._get_shape(node, 0)
+        depth = self._try_get_value(node, 1)
         axis = get_attribute(node, 'axis', -1)
         axis = handle_negative_axis(axis, len(shape)+1)
-        new_shape = shape[:axis] + [self._new_symbolic_dim_from_output(node)] + shape[axis:]
+        new_shape = (shape[:axis] +
+                     [self._new_symbolic_dim_from_output(node) if depth is None else depth] +
```

Comment:
>depth is None [](start = 66, length = 13)

could you check if depth is literal instead? if not is_literal(depth). This might be simpler to achieve what you want when depth is int.
Note that here new_shape is a list of string + int, but _try_get_value may return non-literal sympy symbol or expression and cause helper.make_tensor_value_info to fail if there's sympy expression in the list.
A better fix might be using sympy_shape for input by self._get_sympy_shape, and convert sympy shape to list of string/int via get_shape_from_sympy_shape()

Reply:
Let me know if my new change matches what you meant

Reply:
Looks good, nit: could be better to rename shape to be more explicity: sympy_shape = self._get_sympy_shape(node, 0)

---
In reply to: [464944630](https://github.com/microsoft/onnxruntime/pull/4452#discussion_r464944630) [](ancestors = 464944630)